packages/contract/lib/utils.js

```diff
@@ -255,10 +255,10 @@ const Utils = {
     );
 
     const denominator = bigNumberify(10).pow(significantFigures);
-    const numerator = Math.round(decimal * denominator);
-    const secondOperand = bigNumberify(numerator).div(denominator);
+    const multiplier = Math.round(decimal * denominator);
+    const numerator = bigNumberify(multiplier).mul(bignum);
 
-    return bignum.mul(secondOperand);
+    return numerator.div(denominator);
```

Comment:
So, my algebra is rusty, but I had to reason this out in vim for myself to make sense of it, so thought I'd share:

```
This PR changes this:            To this:

        round( d * 10^P )            n * round( d * 10^P )
v = n * -----------------        v = ---------------------
            10^P                              10^P
```


We're trying to do this:

```
v = n * d
```

Or, expressed as digit placeholders (with underscores as precision gaps):

```
VVVV.0 = NNNN.0 * D.DDD____
```

where:
  - v is the result we want
  - n is `bignum`,
  - d is `decimal`
  - P is `significantFigures`, i.e. some amount of precision

**To [try to?] show how this might play out, the new code does steps:**

1. Scale for desired precision (multiply+round):

   ```
   D.DDD____  ->  DDDD.0
   ```

2. Multiply by our `bingum`:

   ```
   NNNN.0 * DDDD.0 = VVVV____.0
   ```

3. Then divide by original scale:

   ```
   VVVV____.0  ->  VVVV.____
   ```

No loss of precision.


To confirm why the old one was bad, the difference is that we scaled then immediately de-scaled:

1. Scale:
   ```
   D.DDD____  ->  DDDD.0
   ```

2. De-scale:

   ```
   DDDD.0  ->  D.0
   ```

3. Then multiply:
   ```
   NNNN.0 * D.0 = VV__.0
   ```

OK, I'm convinced this is correct and that the old one was not correct.

Reply:
(Should this reasoning be a code comment? Would that spare future incredulity?)

Reply:
I'l leave this convo open for a future Truffler. ⏲